### PR TITLE
Add type annotations for tmt/options.py

### PR DESCRIPTION
**Pull request by thrix**
Signed-off-by: Miroslav Vadkerti <mvadkert@redhat.com>

### CODE REVIEW

**Comment by thrix** [comment]
ready for review

File: tmt/options.py [hunk 1]
```diff
@@ -124,7 +134,7 @@ def _check_method(self, context, args):
             if how is not None:
                 for method in methods:
                     if method.startswith(how):
-                        self._method = methods[method]
+                        self._method = methods[method]  # type: ignore
```
Comment (happz) [inline tmt/options.py:137]: Why this `ignore`?  `methods[method]` would be `click.core.Command`, IIUIC, `_method` has no annotation so far, could it be `Optional[click.core.Command]`?

Comment (thrix) [reply tmt/options.py]: thanks, makes sense, fixed

**Review by janhavlin — APPROVED** [review]

**Comment by thrix** [comment]
@psss rebased to be mergable

**Review by psss — APPROVED** [review]
Thanks, looks good, just removed duplicate `beakerlib.py` from `mypy.ini`.

**Comment by psss** [comment]
Failing `f35` job caused by an unrelated `podman` problem:

    Error: OCI runtime error: /usr/bin/crun: symbol lookup error: /usr/bin/crun: undefined symbol: criu_join_ns_add

Merging.

**Pull request merged**